moshi/src/main/java/com/squareup/moshi/Moshi.java

```diff
@@ -70,6 +71,19 @@
         Collections.singleton(Types.createJsonQualifierImplementation(annotationType)));
   }
 
+  @CheckReturnValue
+  public <T> JsonAdapter<T> adapter(Type type, Class<? extends Annotation>... annotationTypes) {
```

Comment:
in light of my comment on needing the JsonQualifier elements, i'm hoping this isn't needed, at least for this PR.

Reply:
As in you'd prefer to see this API removed? Or that you're hoping needing jsonqualifier elements support isn't needed for this PR?

Reply:
I'm thinking that this API won't be needed because the usage of it won't be the intended behavior (honoring the annotation elements).

Reply:
I don't quite follow. Why doesn't this?